clippy_lints/src/utils/paths.rs

```diff
@@ -90,9 +90,23 @@ pub const PATH_BUF: [&str; 3] = ["std", "path", "PathBuf"];
 pub const PATH_BUF_AS_PATH: [&str; 4] = ["std", "path", "PathBuf", "as_path"];
 pub const PATH_TO_PATH_BUF: [&str; 4] = ["std", "path", "Path", "to_path_buf"];
 pub const POLL: [&str; 4] = ["core", "task", "poll", "Poll"];
+pub const PTR_COPY: [&str; 3] = ["core", "ptr", "copy"];
+pub const PTR_COPY_NONOVERLAPPING: [&str; 3] = ["core", "ptr", "copy_nonoverlapping"];
 pub const PTR_EQ: [&str; 3] = ["core", "ptr", "eq"];
 pub const PTR_NULL: [&str; 3] = ["core", "ptr", "null"];
 pub const PTR_NULL_MUT: [&str; 3] = ["core", "ptr", "null_mut"];
+pub const PTR_READ: [&str; 3] = ["core", "ptr", "read"];
```

Comment:
Seems like this is a perfect lint for https://github.com/rust-lang/rust-clippy/issues/5393. Should we try to add diagnostics to the functions instead of hardcoding them here? I would need some hints how to do this though. :)

Reply:
I think you are right. I'm not familiar with the process of adding diagnostic items, but looking at [this commit](https://github.com/rust-lang/rust/commit/cd159fd7f9f8eec5792e6fd7f1b347ea3e028301) it seems that just adding the attribute should be enough?

This should be done in the rust-lang/rust repo, and we can sync that change afterwards and finish the lint here. As a heads-up we may try to pin to a nightly soon in this repo, so the aforementioned sync process could change. I apologize in advance if that creates any problem, we still have to discover what the new way of working with the pinned nightly will look like :)

Reply:
FYI, there are other lints being currently implemented that will add paths (e.g. #6394)

Even though the right thing to do would be probably to turn those into diagnostic items, I would not make a blocker out of this, and it's fine if you leave the paths here. I understand it would make merging this way longer, and it's not a problem introduced in your PR.

Ideally, you can address #5393 after this one if you want :)

Reply:
You were right about first merging this and then adding diagnostic items to the compiler. I got stuck in the latter step.